.github/workflows/permaref.yaml

```diff
@@ -0,0 +1,40 @@
+# Automatically creates a tag for each commit to `main` so when we rebase
+# changes on top of the upstream, we retain permanent references to each
+# previous commit so they are not orphaned and eventually deleted.
+name: Create permanent reference
+
+on:
+  push:
+    branches:
+      - "main"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Get the permanent ref number
+        id: get_version
+        run: |
+          # Enable pipefail so git command failures do not result in null versions downstream
+          set -x
+
+          echo ::set-output name=LAST_PERMA_NUMBER::$(\
+            git ls-remote --tags --refs --sort="v:refname" \
+            https://github.com/zanieb/pubgrub.git | grep "tags/perma-" | tail -n1 | sed 's/.*\/perma-//' \
+          )
```

Comment:
I think you can do the slightly nicer `echo {name}={value} >> "$GITHUB_OUTPUT"` now.

Reply:
Yeah good note I was about to comment here that I should change to the new syntax.

I copied this from some code I wrote a long time ago to pull versions from tags.

Reply:
> I copied this from some code I wrote a long time ago to pull versions from tags.

We have the exact same GHA development workflow. :P

Reply:
#3